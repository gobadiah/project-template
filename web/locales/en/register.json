{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
