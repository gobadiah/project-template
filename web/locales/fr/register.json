{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
