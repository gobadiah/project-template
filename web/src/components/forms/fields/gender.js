import React from 'react';
import { string } from 'prop-types';

import { Field } from 'redux-form';

import RenderField from './render-field';

export const GenderField = ({ value, label, ...props }) => (
  <Field
    value={value}
    name='gender'
    component={RenderField}
    type='radio'
    label={label}
    {...props}
  />
);

GenderField.propTypes = {
  label: string.isRequired,
  value: string.isRequired,
};

export const MaleField = props => <GenderField value='male' label='Male' {...props} />;
export const FemaleField = props => <GenderField value='female' label='Female' {...props} />;
