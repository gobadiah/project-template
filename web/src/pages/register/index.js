import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
