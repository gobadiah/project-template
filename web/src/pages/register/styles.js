import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
