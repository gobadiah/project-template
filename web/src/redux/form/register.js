import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
