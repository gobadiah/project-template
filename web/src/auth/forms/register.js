import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
