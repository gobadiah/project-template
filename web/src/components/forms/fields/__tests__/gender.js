import 'jsdom-global/register';

import React from 'react';
import { shallow } from 'enzyme';

import { FemaleField, GenderField, MaleField } from '../gender';

describe('FemaleField', () => {
  it('shallow render should match snapshot', () => {
    const wrapper = shallow(<FemaleField />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass down props', () => {
    const wrapper = shallow(<FemaleField some='prop' />);
    expect(wrapper.find('GenderField').props()).toHaveProperty('some', 'prop');
  });
});

describe('MaleField', () => {
  it('shallow render should match snapshot', () => {
    const wrapper = shallow(<MaleField />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass down props', () => {
    const wrapper = shallow(<MaleField some='prop' />);
    expect(wrapper.find('GenderField').props()).toHaveProperty('some', 'prop');
  });
});

describe('GenderField', () => {
  it('shallow render should match snapshot', () => {
    const wrapper = shallow(<GenderField value='trans' label='Trans' this='day' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass some props to Field', () => {
    const wrapper = shallow(<GenderField value='trans' label='Trans' this='day' />);
    const props = wrapper.find('Field').props();
    expect(props).toHaveProperty('value', 'trans');
    expect(props).toHaveProperty('label', 'Trans');
    expect(props).toHaveProperty('this', 'day');
  });
});
