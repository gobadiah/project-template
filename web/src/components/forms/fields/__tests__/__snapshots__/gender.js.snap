// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FemaleField shallow render should match snapshot 1`] = `
<GenderField
  label="Female"
  value="female"
/>
`;

exports[`GenderField shallow render should match snapshot 1`] = `
<Field
  component={[Function]}
  label="Trans"
  name="gender"
  this="day"
  type="radio"
  value="trans"
/>
`;

exports[`MaleField shallow render should match snapshot 1`] = `
<GenderField
  label="Male"
  value="male"
/>
`;
